x-pack/platform/plugins/shared/observability_ai_assistant/server/service/distributed_lock_manager/index.ts

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Client, errors } from '@elastic/elasticsearch';
+import { Logger } from '@kbn/logging';
+import { v4 as uuid } from 'uuid';
+import pRetry from 'p-retry';
+import prettyMilliseconds from 'pretty-ms';
+import { once } from 'lodash';
+
+export const LOCKS_INDEX_ALIAS = '.kibana_locks';
+export const LOCKS_CONCRETE_INDEX_NAME = `${LOCKS_INDEX_ALIAS}-000001`;
+
+export enum LockId {
+  KnowledgeBaseReindex = 'knowledge_base_reindex',
+}
+
+export interface LockDocument {
+  createdAt: string;
+  expiresAt: string;
+  metadata: Record<string, any>;
+  token: string;
+}
+
+interface AcquireOptions {
+  metadata?: Record<string, any>;
+  ttlMs?: number;
+}
+
+const createLocksWriteIndexOnce = once(createLocksWriteIndex);
+
+export class LockManager {
```

Comment:
can we use the Service/Client pattern here that is commonly used across kibana? I'm not a huge fan of expecting the consumer to pass in the ES Client (or the logger).